exo/inference/torch/model/hf.py

```diff
@@ -0,0 +1,274 @@
+import torch
+import torch.nn as nn
+from typing import Tuple, Optional, Union, List
+
+from exo.inference.shard import Shard
+from exo.helpers import DEBUG
+
+from transformers import (
+ AutoModelForCausalLM,
+  DynamicCache,
+  Cache,
+  LogitsProcessorList,
+  TopKLogitsWarper,
+  TopPLogitsWarper,
+  TemperatureLogitsWarper
+)
+
+# llama
+from transformers.models.llama.modeling_llama import LlamaModel
+
+class ShardedHuggingFaceModel:
+  def __init__(
+    self,
+    shard: Shard,
+    local_model_path,
+    device,
+    dtype,
+    top_k: int = 25,
+    temp: float = 0.7,
+    top_p: float = 0.9,
+    max_length: int = 50,
+    max_time: float = 10.0
+  ):
+    # class vars
+    self.shard = shard
+    self.hidden_states = None
+    self.input_ids = None
+    self.inputs_embeds = None
+    self.attention_mask = None
+    self.position_embeddings = None
+    self.past_key_values = None
+    self.cache_position = None
+    self.position_ids = None
+    self.causal_mask = None
+    self.local_model_path = local_model_path
+
+    # setup logit processors
+    self.logits_processor = LogitsProcessorList([
+      TopKLogitsWarper(top_k),
+      TemperatureLogitsWarper(temp),
+      TopPLogitsWarper(top_p)
+    ])
+
+    self.device = device
+    self.torch_dtype = dtype
+
+    # setup pytorch and transformer llm
+    try:
+      self.llm_model = AutoModelForCausalLM.from_pretrained(
+        pretrained_model_name_or_path=self.local_model_path,
+        torch_dtype=self.torch_dtype,
+        device_map={"": self.device},
```

Comment:
@risingsunomi I just changed this - can you double check this is fine? Before it was not working when I set `TORCH_DEVICE=cpu` on my Mac

Reply:
settings
```
device_map={"": self.device}
```
Gives me an error. 

set' object is not subscriptable so set it back to "auto" for now